bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/validation/SitemapValidator.xtend

```diff
@@ -21,66 +21,90 @@ import org.eclipse.smarthome.model.sitemap.Sitemap
+                return;
+            }
+        }
+    }
 
-	@Check
-	def void checkFramesInWidgetList(Sitemap sitemap) {
-		var containsFrames = false
-		var containsOtherWidgets = false
+    @Check
+    def void checkFramesInWidgetList(Sitemap sitemap) {
+        var containsFrames = false
+        var containsOtherWidgets = false
 
-		for (Widget w : sitemap.children) {
-			if (w instanceof Frame) {
-				containsFrames = true
-			} else {
-				containsOtherWidgets = true
-			}
-			if (containsFrames && containsOtherWidgets) {
-				error("Sitemap should contain either only frames or none at all",
-					SitemapPackage.Literals.SITEMAP.getEStructuralFeature(SitemapPackage.SITEMAP__NAME));
-				return
-			}
-		}
-	}
+        for (Widget w : sitemap.children) {
+            if (w instanceof Frame) {
+                containsFrames = true
+            } else {
+                containsOtherWidgets = true
+            }
+            if (containsFrames && containsOtherWidgets) {
+                error("Sitemap should contain either only frames or none at all",
+                    SitemapPackage.Literals.SITEMAP.getEStructuralFeature(SitemapPackage.SITEMAP__NAME));
+                return
+            }
+        }
+    }
 
-	@Check
-	def void checkFramesInWidgetList(LinkableWidget widget) {
-		if (widget instanceof Frame) {
+    @Check
+    def void checkFramesInWidgetList(LinkableWidget widget) {
+        if (widget instanceof Frame) {
 
-			// we have a dedicated check for frames in place
-			return;
-		}
-		var	containsFrames = false
-		var	containsOtherWidgets = false
-		for (Widget w : widget.children) {
-			if (w instanceof Frame) {
-				containsFrames = true
-			} else {
-				containsOtherWidgets = true
-			}
-			if (containsFrames && containsOtherWidgets) {
-				error("Linkable widget should contain either only frames or none at all",
-					SitemapPackage.Literals.FRAME.getEStructuralFeature(SitemapPackage.LINKABLE_WIDGET__CHILDREN));
-				return
-			}
-		}
-	}
+            // we have a dedicated check for frames in place
+            return;
+        }
+        var containsFrames = false
+        var containsOtherWidgets = false
+        for (Widget w : widget.children) {
+            if (w instanceof Frame) {
+                containsFrames = true
+            } else {
+                containsOtherWidgets = true
+            }
+            if (containsFrames && containsOtherWidgets) {
+                error("Linkable widget should contain either only frames or none at all",
+                    SitemapPackage.Literals.FRAME.getEStructuralFeature(SitemapPackage.LINKABLE_WIDGET__CHILDREN));
+                return
+            }
+        }
+    }
 
+    @Check
+    def void checkSetpoints(Setpoint sp) {
+        if (sp.item === null) {
+            error("Setpoint without an item defined",
+                SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT));
+        }
+
+        if (BigDecimal.ZERO == sp.step) {
```

Comment:
should this also be compared by `===`?

Reply:
No!

* `===` is checking for object identity
* `==` is using the equals implementation

Reply:
thanks!